### PR TITLE
Support ember http/2 over unix sockets

### PR DESCRIPTION
**Pull request by armanbilge**
Most of the changes are shoe-horning an `Either[UnixSocketAddress, SocketAddress[Host]]` into the APIs, because we are lacking a universal datatype to represent these. See https://github.com/Comcast/ip4s/issues/466.

### CODE REVIEW

File: ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Server.scala [hunk 1]
```diff
@@ -194,8 +195,10 @@ private[ember] object H2Server {
       F.sleep(1.seconds) >> stateRef.get.map(_.closed).ifM(F.unit, holdWhileOpen(stateRef))
 
     def initH2Connection: F[H2Connection[F]] = for {
-      address <- socket.remoteAddress
-      (remotehost, remoteport) = (address.host, address.port)
+      address <- socket.remoteAddress.attempt.map(
+        // TODO, only used for logging
+        _.leftMap(_ => UnixSocketAddress("unknown.sock"))
+      )
```
Comment (armanbilge) [inline ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Server.scala:201]: Yeah, I'm sort of flaking on this. /shrug

File: ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Server.scala [hunk 2]
```diff
@@ -194,8 +195,10 @@ private[ember] object H2Server {
       F.sleep(1.seconds) >> stateRef.get.map(_.closed).ifM(F.unit, holdWhileOpen(stateRef))
 
     def initH2Connection: F[H2Connection[F]] = for {
-      address <- socket.remoteAddress
-      (remotehost, remoteport) = (address.host, address.port)
+      address <- socket.remoteAddress.attempt.map(
+        // TODO, only used for logging
+        _.leftMap(_ => UnixSocketAddress("unknown.sock"))
```
Comment (ChristopherDavenport) [inline ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Server.scala:200]: Hmmmm. This is definitely weird.

File: ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Server.scala (hunk 1 above)
Comment (ChristopherDavenport) [reply ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Server.scala:201]: Hmmmm. This is definitely weird.

Comment (danicheg) [reply ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Server.scala:201]: 😮

Comment (armanbilge) [reply ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Server.scala:201]: Yeah, so for the record the upstream issue is here.
- https://github.com/Comcast/ip4s/issues/466

**Review by ChristopherDavenport — APPROVED** [review]

**Pull request merged**